### ISSUE TITLE
Normals estimation

### ISSUE THREAD

**Issue by Ilyabasharov** [nerfstudio-project/nerfstudio#1597]
Hello, thank you for the great work! 
I have question about normals estimation.

I think sign of predicted normals are not correct. `FieldHeadNames.NORMALS` [here](https://github.com/nerfstudio-project/nerfstudio/blob/630babacd6aea32999f40c181b7e8857d9a483b4/nerfstudio/fields/base_field.py#L94) and `FieldHeadNames.PRED_NORMALS` [here](https://github.com/nerfstudio-project/nerfstudio/blob/630babacd6aea32999f40c181b7e8857d9a483b4/nerfstudio/field_components/field_heads.py#L204) have different signs. I think it is not correct but maybe i'm wrong. What do you think?

**Comment by tancik** [comment]
The predicted normals are the output to a network and are trained to predict the actual normals. The network can learn to predict positive or negative values, explicitly putting a negative here shouldn't make a difference.

**Comment by Ilyabasharov** [comment]
@tancik thank you! I was asking you because for future methods like Ref-NeRF it is important to make convergence better.

**Comment by hermosaaurora** [comment]
I'm curious about the function, why can it estimate the normals of the surface? Do you have any paper that supports that?

https://github.com/nerfstudio-project/nerfstudio/blob/630babacd6aea32999f40c181b7e8857d9a483b4/nerfstudio/fields/base_field.py#L75-L96

**Comment by maturk** [comment]
> I'm curious about the function, why can it estimate the normals of the surface? Do you have any paper that supports that?
> 
> https://github.com/nerfstudio-project/nerfstudio/blob/630babacd6aea32999f40c181b7e8857d9a483b4/nerfstudio/fields/base_field.py#L75-L96

Here they take the gradient of the density field (wrt x), which is an approximation of the normal direction (although a bit noisy) which was introduced in ref-nerf. The intuition is as follows: imagine a solid surface, before the surface density is low, at the surface density is high, and beyond the surface density is low again. The normal direction will thus be proportional to the (negative) gradient of the density field.

**Comment by hermosaaurora** [comment]
hi thank you for your reply! I also noticed one paper [UNISURF](https://arxiv.org/pdf/2104.10078.pdf), which also has similar ideas about how to estimate surface normals.

<img width="460" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/142530595/1abb940b-1ead-4cd7-a7cb-540ab0b3637a">

<img width="462" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/142530595/fb581cb9-a1d7-428f-8070-85567210b144">


But in this paper, it calculates the normal from $o_\theta$, and from section4.1 and section4.2 I think it assumes the normals should be calculated by taking the gradients of $\alpha$, but in Ref-NeRF we obtain the normals by taking the gradients of $\sigma$ (also density)

**Comment by maturk** [comment]
> hi thank you for your reply! I also noticed one paper [UNISURF](https://arxiv.org/pdf/2104.10078.pdf), which also has similar ideas about how to estimate surface normals.
> 
> <img alt="image" width="460" src="https://user-images.githubusercontent.com/142530595/262745976-1abb940b-1ead-4cd7-a7cb-540ab0b3637a.png"> <img alt="image" width="462" src="https://user-images.githubusercontent.com/142530595/262746292-fb581cb9-a1d7-428f-8070-85567210b144.png">
> But in this paper, it calculates the normal from oθ, and from section4.1 and section4.2 I think it assumes the normals should be calculated by taking the gradients of α, but in Ref-NeRF we obtain the normals by taking the gradients of σ (also density)

Right, but this is occupancy right? I think it is the same idea though, if occupancy goes from 0 to 1 and then 0 again across a ray/segment, then the normal direction is the gradient of this. You could do this across a surface and get the normals. Is my understanding correct?
